include/s3select.h

```diff
@@ -2448,17 +2521,27 @@ class base_s3object
 	for (auto& i : m_projections)
 	{
 	  projections_resuls.push_value( &(i->eval()) );
-	}
-	result_values_to_string(projections_resuls,result);
-	if(m_sql_processing_status == Status::SQL_ERROR)
-	{
-	  return m_sql_processing_status; 
-	}
+    col_ref.push_back(i->is_column());
+    if(is_json_query && !is_star && i->is_column())  {
+      last_key = m_sa->get_json_key_path();
+    } else {
+      last_key.push_back("");
       }
```

Comment:
the above routine repeats many times.
and actually will produce the same result each time.

thus, it can be executed only once.